### PR TITLE
mod : Armor impact Cavalry , No More Sliding , reduced impact of speed on damage for highspeed

### PR DESCRIPTION
**Pull request by namidaka**

### CODE REVIEW

File: src/Module.Server/Common/Models/CrpgAgentStatCalculateModel.cs [hunk 1]
```diff
@@ -224,13 +224,13 @@ private void UpdateMountAgentStats(Agent agent, AgentDrivenProperties props)
         int ridingSkill = agent.RiderAgent != null
             ? GetEffectiveSkill(agent.RiderAgent.Character, agent.RiderAgent.Origin, agent.RiderAgent.Formation, DefaultSkills.Riding)
             : 100;
+        int harnessArmor = mountHarness.Item?.ArmorComponent?.BodyArmor ?? 0;
+        float armoredPercentage = harnessArmor / 60f;
+        float armorImpactOnSpeed = 1f / (1f + 0.6f * armoredPercentage);
         props.MountManeuver = mount.GetModifiedMountManeuver(in mountHarness) * (0.5f + ridingSkill * 0.0025f);
-        props.MountSpeed = (mount.GetModifiedMountSpeed(in mountHarness) + 1) * 0.33f * (1.0f + ridingSkill * 0.0008f);
+        props.MountSpeed = (mount.GetModifiedMountSpeed(in mountHarness) + 1) * 0.3f * (float)(0.55f + ridingSkill * 0.0008f + 1 / (2.2f + Math.Pow(2, -0.08f * (ridingSkill - 70f)))) * armorImpactOnSpeed; // speed reduced by 37.5% for full armor
```
Comment (verdie-g) [inline src/Module.Server/Common/Models/CrpgAgentStatCalculateModel.cs:231]: Can you break that into several lines

File: src/Module.Server/Common/Models/CrpgStrikeMagnitudeModel.cs [hunk 2]
```diff
@@ -65,11 +65,11 @@ public override float GetBluntDamageFactorByDamageType(DamageTypes damageType)
             case WeaponClass.TwoHandedPolearm:
             case WeaponClass.LowGripPolearm:
                 impactPointFactor = (float)Math.Pow(10f, -4f * Math.Pow(impactPoint - 0.93, 2f));
-                return BladeDamageFactorToDamageRatio * (0.4f + 0.6f * impactPointFactor) * ((float)Math.Pow(swingSpeedPercentage, 5f) + extraLinearSpeed / 10f);
+                return BladeDamageFactorToDamageRatio * (0.4f + 0.6f * impactPointFactor) * (float)(Math.Pow(swingSpeedPercentage, 5f) + Math.Pow(extraLinearSpeed / 20f, 0.7f) + Math.Pow(extraLinearSpeed / 22f, 4f));
```
Comment (verdie-g) [inline src/Module.Server/Common/Models/CrpgStrikeMagnitudeModel.cs:68]: Same here

File: src/Module.Server/Common/Models/CrpgStrikeMagnitudeModel.cs [hunk 3]
```diff
@@ -65,11 +65,11 @@ public override float GetBluntDamageFactorByDamageType(DamageTypes damageType)
...
 
             default: // Weapon that do not have a wooden handle
                 impactPointFactor = (float)Math.Pow(10f, -4f * Math.Pow(impactPoint - 0.75, 2f));
-                return BladeDamageFactorToDamageRatio * (0.8f + 0.2f * impactPointFactor) * ((float)Math.Pow(swingSpeedPercentage, 5f) + extraLinearSpeed / 10f);
+                return BladeDamageFactorToDamageRatio * (0.8f + 0.2f * impactPointFactor) * (float)(Math.Pow(swingSpeedPercentage, 5f) + Math.Pow(extraLinearSpeed / 20f, 0.7f) + Math.Pow(extraLinearSpeed / 22f, 4f));
```
Comment (verdie-g) [inline src/Module.Server/Common/Models/CrpgStrikeMagnitudeModel.cs:72]: Same here

File: src/Module.Server/Common/Models/CrpgStrikeMagnitudeModel.cs [hunk 4]
```diff
@@ -84,13 +84,14 @@ public override float GetBluntDamageFactorByDamageType(DamageTypes damageType)
...
+        float extraLinearSpeedSign = Math.Sign(extraLinearSpeed);
         switch (weaponUsageComponent.WeaponClass)
         {
             case WeaponClass.OneHandedSword:
             case WeaponClass.Dagger:
-                 return BladeDamageFactorToDamageRatio * ((float)Math.Pow(thrustSpeedPercentage, 2f) + extraLinearSpeed / 10f);
+                 return BladeDamageFactorToDamageRatio * (float)(Math.Pow(thrustSpeedPercentage, 2f) + extraLinearSpeedSign * Math.Pow(extraLinearSpeedSign * extraLinearSpeed / 20f, 0.7f) + extraLinearSpeedSign * Math.Pow(extraLinearSpeed / 22f, 4f));
             default:
-                 return BladeDamageFactorToDamageRatio * ((float)Math.Pow(thrustSpeedPercentage, 6f) + extraLinearSpeed / 10f);
+                 return BladeDamageFactorToDamageRatio * (float)(Math.Pow(thrustSpeedPercentage, 6f) + extraLinearSpeedSign * Math.Pow(extraLinearSpeedSign * extraLinearSpeed / 20f, 0.7f) + extraLinearSpeedSign * Math.Pow(extraLinearSpeed / 22f, 4f));
```
Comment (verdie-g) [inline src/Module.Server/Common/Models/CrpgStrikeMagnitudeModel.cs:94]: Could you factorize this code

File: src/Module.Server/Common/Models/CrpgAgentStatCalculateModel.cs [hunk 5]
```diff
@@ -224,13 +224,13 @@ private void UpdateMountAgentStats(Agent agent, AgentDrivenProperties props)
         int ridingSkill = agent.RiderAgent != null
             ? GetEffectiveSkill(agent.RiderAgent.Character, agent.RiderAgent.Origin, agent.RiderAgent.Formation, DefaultSkills.Riding)
             : 100;
+        int harnessArmor = mountHarness.Item?.ArmorComponent?.BodyArmor ?? 0;
+        float armoredPercentage = harnessArmor / 60f;
```
Comment (verdie-g) [inline src/Module.Server/Common/Models/CrpgAgentStatCalculateModel.cs:228]: I guess 60 is the max armor, you could move that into a constant
```
const float maxHarnessArmor = 60f;
```

**Review by verdie-g — CHANGES_REQUESTED** [review]

File: src/Module.Server/Common/Models/CrpgStrikeMagnitudeModel.cs (hunk 2 above)
Comment (namidaka) [reply src/Module.Server/Common/Models/CrpgStrikeMagnitudeModel.cs]: done

File: src/Module.Server/Common/Models/CrpgStrikeMagnitudeModel.cs (hunk 3 above)
Comment (namidaka) [reply src/Module.Server/Common/Models/CrpgStrikeMagnitudeModel.cs]: done

File: src/Module.Server/Common/Models/CrpgStrikeMagnitudeModel.cs (hunk 4 above)
Comment (namidaka) [reply src/Module.Server/Common/Models/CrpgStrikeMagnitudeModel.cs]: done

File: src/Module.Server/Common/Models/CrpgAgentStatCalculateModel.cs (hunk 5 above)
Comment (namidaka) [reply src/Module.Server/Common/Models/CrpgAgentStatCalculateModel.cs]: done

File: src/Module.Server/Common/Models/CrpgAgentStatCalculateModel.cs (hunk 1 above)
Comment (namidaka) [reply src/Module.Server/Common/Models/CrpgAgentStatCalculateModel.cs]: done

**Review by verdie-g — APPROVED** [review]

**Pull request merged**